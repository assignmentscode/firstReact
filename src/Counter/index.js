import React, { Component } from 'react';

class Counter extends Component {
  render(){
    return (
      <div className="Counter">
      <p>Counter : 0</p>
      <span>
      <button type="button">Add</button>
      <button type="button">Minus</button>
      </span>
      </div>
    );
    }
}
export default Counter;