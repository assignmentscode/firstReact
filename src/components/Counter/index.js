import React, { Component } from 'react';
import './App.css';
import Button from './../Button/index';

class Counter extends Component {
  state = {
    counter : this.props.initialValue,
  }
  increamentCounter = () => this.setState({counter: this.state.counter + 1});
  decreamentCounter = () => this.setState({counter: this.state.counter - 1});
  render(){
    return (
      <div className="Counter">
      <p>Counter : {this.state.counter}</p>
      <span>
      <Button buttonText="Increment" onClick={this.increamentCounter}/>
      <Button buttonText="Decreament" onClick={this.decreamentCounter}/>
      </span>
      </div>
    );
    }
}
export default Counter;