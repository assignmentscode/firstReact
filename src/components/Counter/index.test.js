import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Counter from './index';

describe('Counter', () =>{
  it('should render a  button', () => {
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should increment the initial Value of counter', () =>{
    const wrapper = shallow(<Counter initialValue={11} />);
    wrapper.instance().increamentCounter();
    expect(wrapper.instance().state.counter).toEqual(12);
  });
  it('should decreament the initial Value of counter', () =>{
    const wrapper = shallow(<Counter initialValue={0} />);
    wrapper.instance().decreamentCounter();
    expect(wrapper.instance().state.counter).toEqual(-1);
  });
  it('should check that initial Value of counter to be same as passed', () =>{
    const wrapper = shallow(<Counter initialValue={5} />);
    expect(wrapper.instance().state.counter).toEqual(5);
  });
});