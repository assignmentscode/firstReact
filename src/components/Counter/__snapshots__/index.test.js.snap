// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render a  button 1`] = `
<div
  className="Counter"
>
  <p>
    Counter : 
  </p>
  <span>
    <button
      onClick={[Function]}
      type="button"
    >
      Increment
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Decreament
    </button>
  </span>
</div>
`;
